### PR TITLE
Add @portrait endpoint

### PR DESCRIPTION
**Pull request by sneridagh**
Fixes: #1480 and addresses https://github.com/plone/volto/pull/3487 (the /portal_memberdata/portrait traversal is not working with ++api++).

Let me know what do you think.

/cc @jmevissen

### CODE REVIEW

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by sneridagh** [comment]
Needs docs.

**Comment by sneridagh** [comment]
Strange, it fails here:
https://github.com/plone/plone.restapi/runs/8130021161?check_suite_focus=true#step:12:37

```
diff --git a/src/plone/restapi/tests/http-examples/translated_messages_addons.resp b/src/plone/restapi/tests/http-examples/translated_messages_addons.resp
--- a/src/plone/restapi/tests/http-examples/translated_messages_addons.resp
+++ b/src/plone/restapi/tests/http-examples/translated_messages_addons.resp
@@ -73,7 +73,7 @@ Content-Type: application/json
                 "newVersion": "0006",
                 "required": false
             },
-            "version": "8.24.2.dev0"
+            "version": "8.25.1.dev0"
         },
         {
             "@id": "http://localhost:55001/plone/@addons/plone.session",
```

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/serializer/user.py [hunk 1]
```diff
@@ -42,9 +42,7 @@ def __call__(self):
                 memberdata = getToolByName(portal, "portal_memberdata")
                 safe_id = membership._getSafeMemberId(user.id)
                 if safe_id in memberdata.portraits:
-                    value = "{}/portal_memberdata/portraits/{}".format(
-                        portal.absolute_url(), safe_id
-                    )
+                    value = "{}/@portrait/{}".format(portal.absolute_url(), safe_id)
```
Comment (ericof) [inline src/plone/restapi/serializer/user.py:45]: Use f-strings instead of format

**Review by ericof — APPROVED** [review]
Just a small comment about usage of f-strings

File: src/plone/restapi/serializer/user.py (hunk 1 above)
Comment (sneridagh) [reply src/plone/restapi/serializer/user.py]: Done

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Review by erral — APPROVED** [review]
- What about setting/deleting the portrait? a PATCH/DELETE endpoint could be added to set the portrait, right? Or is it handled by the `@user` endpoint?

- It could also be interesting to add an additional property `portrait` to the `@user` endpoint signaling that the portrait is in this endpoint.

**Comment by sneridagh** [comment]
@erral 

> * What about setting/deleting the portrait? a PATCH/DELETE endpoint could be added to set the portrait, right? Or is it handled by the `@user` endpoint?

It's handled by the @user endpoint (as it's a user property). We could also do it here, but might be redundant. Something to talk about it.

> * It could also be interesting to add an additional property `portrait` to the `@user` endpoint signaling that the portrait is in this endpoint.

I've changed the serialization for users, so the @user GET returns the @portrait URL. I hope that this makes the roundtrip. 

The important thing was to fix the ++api++ traversal problem with the portraits.

File: docs/source/portrait.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+html_meta:
+  "description": "Returns user's portrait in a Plone site using the @portrait endpoint"
+  "property=og:description": "Returns user's portrait in a Plone site using the @portrait endpoint"
+  "property=og:title": "Portraits"
+  "keywords": "Plone, plone.restapi, REST, API, Users, portrait, profile"
+---
+
+# Portraits
+
+A Plone users have the option to upload a portrait to their profile. This profile is used in several occassions in Plone's user interface like in the comments view.
```
Comment (davisagli) [inline docs/source/portrait.md:11]:
```suggestion
Plone users have the option to upload a portrait to their profile. This profile is used in several places in Plone's user interface, like in the comments view.
```

File: docs/source/portrait.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+## Get self portrait
+
+You can request your own user portrait by issuing a `GET` request to the root URL appending `/@portrait`:
+
+```{eval-rst}
+..  http:example:: curl httpie python-requests
+    :request: ../../src/plone/restapi/tests/http-examples/portrait_self_get.req
+```
+
+The server will respond with a `Status 200` and the image requested (not JSON). The content type is set accordingly. One can use it directly into HTML `src` properties:
```
Comment (davisagli) [inline docs/source/portrait.md:22]:
```suggestion
The server will respond with a `Status 200` and the image requested (not JSON). The content type is set accordingly. One can use it directly in HTML `src` properties:
```

File: docs/source/portrait.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    :request: ../../src/plone/restapi/tests/http-examples/portrait_self_get.req
+```
+
+```{literalinclude} ../../src/plone/restapi/tests/http-examples/portrait_self_404_get.resp
+:language: http
+```
+
+## Get specific user portrait
+
+You can request for and specific user `username` portrait by issuing a `GET` request to the root URL appending `/@portrait/username`:
```
Comment (davisagli) [inline docs/source/portrait.md:41]:
```suggestion
You can request the portrait for a specific user `username` by issuing a `GET` request to the root URL appending `/@portrait/username`:
```

File: docs/source/portrait.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+```{eval-rst}
+..  http:example:: curl httpie python-requests
+    :request: ../../src/plone/restapi/tests/http-examples/portrait_username_get.req
+```
+
+The server will respond with a `Status 200` and the image requested (not JSON). The content type is set accordingly. One can use it directly into HTML `src` properties:
```
Comment (davisagli) [inline docs/source/portrait.md:48]:
```suggestion
The server will respond with a `Status 200` and the image requested (not JSON). The content type is set accordingly. One can use it directly in HTML `src` properties:
```

**Review by davisagli — APPROVED** [review]
I suggested some edits to the docs. Otherwise LGTM.

In the longer term I think we should look into providing image scales for user portraits in the same format as image fields, but that is a bigger project and this looks like a good improvement in the meantime.

**Pull request merged**